docs/DeveloperGuide.md

```diff
@@ -1,38 +1,337 @@
+5. `EditCommand` first gets the relevant `Module` and invokes `editTaskFromModule(targetModule)`.
+6. `editTaskFromModule(targetModule)` retrieves the task `targetTask` specified by the index and invokes `targetTask.setTaskDescription(description)` to change the description.
+
+<br>
+
+### 6.2. Tag Feature
+
+The tag feature allows the user to add user-created one-word tags to each task, so that tasks can be filtered for easily. Each task stores its tags in an `ArrayList<String>`.
+
+The following sequence diagram illustrates the process:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/TagSeqDiagrams/Tag.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/TagSeqDiagrams/GetModule.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/TagSeqDiagrams/CheckAndRunTagOperation.puml)
+
+Here is an example on adding a tag to a general task:  
+
+1. User inputs `tag add 2 testTag`.
+2. `ModHappyParser` identifies the command word as `tag` and passes `add 2 "testTag"` to `TagParser`.
+3. `TagParser` instantiates a `TagCommand` with `tagOperation = "add"`, `taskIndex = 2`, `tagDescription = "testTag"` and `taskModule = null`. This is returned to `Main`.
+4. `Main` calls the `execute()` method of the `TagCommand` instance.
+5. `TagCommand` first gets the relevant `Module`. Since `taskModule` is null, `getGeneralTasks()` is called and the General Tasks `Module` is retrieved.
+6. Next, `TagCommand` checks the `tagOperation`. As its value is `add`, `addTag(targetModule)` is called.
+7. Finally, command feedback is returned to `Main`, indicating that the operation was successful.
+
+<br>
+
+### 6.3. Grade Feature
+
+The Grade feature allows the user to input their predicted/actual grade, according to the official grades that NUS supports.
+
+Here is an example on how to assign a grade to a module:
+
+1. User inputs `grade CS2113T A+`
+2. `ModHappyParser` identifies the command word as `grade` and passes `CS2113T A+` to `GradeParser`.
+3. `GradeParser` instantiates a `GradeCommand` with `moduleCode = "CS2113T"`, `moduleGrade = "A+"`. This is returned to `Main`.
+4. `Main` calls the `execute()` method of the `GradeCommand` instance.
+5. `execute()` retrieves the `Module` instance of `CS2113T` if it exists and invokes `addGradeToModule(m)`.
+6. `addGradeToModule(m)` then invokes `m.setModuleGrade(moduleGrade)` to assign the input grade to the specified module.
+
+Below is the sequence diagram of how the Grade feature works:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/Grade.puml)
```

Comment:
![image](https://user-images.githubusercontent.com/65173208/161184608-ef5a489a-df9a-4c57-9c87-0bf518280550.png)

I'm not too sure about this, but should the X for end of lifeline be below the return statement rather than on the statement itself?